src/unstract/sdk/llm.py

```diff
@@ -251,6 +252,11 @@ def run_completion(
                 raise RateLimitError(msg)
             raise LLMError(msg) from e
 
+    def is_noOp(self, llm: LlamaIndexLLM) -> bool:
```

Comment:
@harini-venkataraman why is this necessary in the base class? Its similar to defining a function to check if its of type `OpenAILLM`

Reply:
@chandrasekharan-zipstack This method is introduced since we needed to identify the NoOp adapter to differentiate JSON type response for enforce type like JSON, table and record.

Reply:
@harini-venkataraman its not a good practice to include such a method in the base class implementation. Do you have ideas on how we can avoid it?

Reply:
@chandrasekharan-zipstack I understand that checking a specific adapter instance is a very targeted approach in the base class. To enforce types like Table, we utilize the details returned for subsequent operations, which may ultimately raise an exception. Instead, we might need to  handle this by checking whether we are using the no-op adapter and return early if so.

While this does not necessarily require the inclusion of a new method in the base class, we can perform the necessary checks using existing getters within the backend/prompt_service. It seems more appropriate to keep the adapter logic within the LLM's base class, rather than integrating it into this flow. For reference, please see the relevant code [here](https://github.com/Zipstack/unstract-cloud/blob/4115532bb9a17683e30eb17b81c2a69f7f679384/prompt_service/plugins/table-extractor/src/base.py#L41).
CC : @hari-kuriakose

Reply:
@chandrasekharan-zipstack  Considering the NoOps, added a check in the flow and removed from here. Thanks for the suggestion.